### PR TITLE
Add/Update copilot instruction

### PR DESCRIPTION
**Pull request by yeelam-gordon**
## Summary of the Pull Request
This pull request significantly updates and expands the documentation and contributor guidelines for the WSL repository, with a focus on improving clarity, discoverability, and support for GitHub Copilot agents and skills. The main `.github/copilot-instructions.md` file has been restructured for easier navigation and updated best practices, while two new instruction files provide standards for creating custom Copilot agents and agent skills specific to WSL development.

**Major documentation overhaul and new contributor guidelines:**

### 1. Documentation Restructuring and Best Practices
- The `.github/copilot-instructions.md` file has been rewritten for clarity, with sections reorganized for easier access to documentation links, build/test/debug workflows, validation steps, repository navigation, and timing guidelines. The new format emphasizes best practices, validation checklists, and cross-platform development notes.

### 2. Copilot Agent and Skill Authoring Standards

**Agent Skills Guidelines**
- Added `.github/instructions/agent-skills.instructions.md`, detailing how to create Agent Skills for Copilot in the WSL repo. Includes required SKILL.md format, frontmatter, bundling resources, and content best practices.

**Custom Agent Guidelines**
- Added `.github/instructions/agents.instructions.md`, providing standards for creating custom Copilot agent files. Covers frontmatter properties, tool configuration, prompt structure, handoff workflows, and sub-agent invocation.

### 3. Improved Navigation and Scoped Instructions
- The main instructions file now includes a table of all scoped instruction files, explaining when and how each applies (e.g., for C++, C#, build, test, debugging, documentation, distributions, pipelines, etc.), making it easier for contributors to find relevant guidelines.

---

**References:**  
- Documentation restructuring and best practices (.github/copilot-instructions.md)  
- Agent Skills guidelines (.github/instructions/agent-skills.instructions.md)  
- Custom Agent guidelines (.github/instructions/agents.instructions.md)

## PR Checklist

- [x] **Communication:** I've discussed this with core contributors already. If work hasn't been agreed, this work might be rejected
- [x] **Tests:** Added/updated if needed and all pass
- [x] **Dev docs:** Added/updated if needed

### CODE REVIEW

File: .github/prompts/create-pr-summary.prompt.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+---
+agent: 'agent'
+description: 'Generate a PowerToys-ready pull request description from the local diff'
+---
+
+# Generate PR Summary
+
+**Goal:** Produce a ready-to-paste PR title and description that follows PowerToys conventions by comparing the current branch against a user-selected target branch.
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/prompts/create-pr-summary.prompt.md]: Fixed: Replaced all PowerToys-specific wording, conventions, and terminology with WSL-specific content (WSL components, paths, validation steps, PR checklist items).

File: .github/prompts/create-commit-title.prompt.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+  ```@terminal
+  git diff HEAD
+  ```
+
+## How to decide the title
+1. From the diff, find the dominant area (e.g., `src/modules/*`, `doc/devdocs/**`) and the change type (bug fix, docs update, config tweak).
+2. Draft an imperative, plain-ASCII title that:
+   - Mentions the primary component when obvious (e.g., `FancyZones:` or `Docs:`)
+   - Stays within 80 characters and has no trailing punctuation
+
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/prompts/create-commit-title.prompt.md:22]: Fixed: Replaced PowerToys paths (src/modules/*, FancyZones) with WSL-relevant scopes (src/windows/service/, src/linux/init/, src/shared/, doc/, distributions/, etc.).

File: .github/prompts/create-commit-title.prompt.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+**Summary rules**
+- Imperative, present tense (“add”, “update”, “remove”, “fix”)
+- Keep it <= 72 characters when possible; be specific, avoid “misc changes”
+
+**Examples**
+- `feat(fancyzones): add canvas template duplication`
+- `fix(mouseutils): guard crosshair toggle when dpi info missing`
+- `docs(runner): document tray icon states`
+- `build(installer): align wix v5 suffix flag`
+- `ci(ci): cache pipeline artifacts for x64`
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/prompts/create-commit-title.prompt.md]: Fixed: Replaced PowerToys scope list with WSL components: service, wsl, wslhost, wslrelay, wslg, libwsl, wslsettings, init, gns, plan9, localhost, relay, drvfs, docs, distributions, pipelines, build, test, config.

File: .github/instructions/cpp.instructions.md [hunk 4]
```diff
@@ -0,0 +1,66 @@
+---
+description: 'C/C++ coding standards, formatting, and copyright headers for WSL source code'
+applyTo: '**/*.{cpp,h,c}'
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/cpp.instructions.md]: Fixed: Changed applyTo from brace expansion `**/*.{cpp,h,c}` to comma-separated globs `**/*.cpp, **/*.h, **/*.c` for consistency with repo instruction-file guidelines.

File: .github/prompts/create-pr-summary.prompt.md [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+**Repo guardrails:**
+- Treat `.github/pull_request_template.md` as the single source of truth; load it at runtime instead of embedding hardcoded content in this prompt.
+- Preserve section order from the template but only surface checklist lines that are relevant for the detected changes, filling them with `[x]`/`[ ]` as appropriate.
+- Cite touched paths with inline backticks, matching the guidance in `.github/copilot-instructions.md`.
+- Call out validation explicitly: list automated checks run (for example TAEF, Linux unit tests, `mkdocs build -f doc/mkdocs.yml`, or `python3 distributions/validate.py`) or state why they are not applicable.
+- Use WSL terminology when describing impacted components, such as `wsl.exe`, `wslservice`, `wslhost`, `init`, `gns`, `plan9`, `relay`, `drvfs`, `wslsettings`, `doc/`, or `distributions/`.
+
+**Workflow:**
+1. Determine the target branch from user context; default to `master` when no branch is supplied.
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/prompts/create-pr-summary.prompt.md]: Fixed: Now resolves default branch dynamically via git symbolic-ref refs/remotes/origin/HEAD, falling back to master.

File: .github/instructions/build-deploy.instructions.md [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+| MSI install | `bin\<platform>\<target>\wsl.msi` |
+| Deploy script | `powershell tools\deploy\deploy-to-host.ps1` |
+| Hyper-V VM | `powershell tools\deploy\deploy-to-vm.ps1 -VmName <vm> -Username <user> -Password <pass>` |
+
+## Timeouts
+
+| Operation | Duration | Set Timeout |
+|-----------|----------|-------------|
+| Full build | 20–45 min | 60+ min |
+| NEVER cancel a build in progress. |
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/build-deploy.instructions.md]: Fixed: Moved 'NEVER cancel' warning out of table into a blockquote to fix malformed row.

File: .github/instructions/cpp.instructions.md [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+
+## Pre-Commit Checklist
+
+1. `clang-format --dry-run --style=file` on changed C/C++ files
+2. `python3 tools/devops/validate-copyright-headers.py` (ignore `_deps/` warnings)
+3. Full Windows build if core components changed
+
+## Key Libraries
+
+| Library | Version | Usage |
```
Comment (benhillis) [inline .github/instructions/cpp.instructions.md]: we should not have the version numbers in here, these get updated periodically

Comment (yeelam-gordon) [reply .github/instructions/cpp.instructions.md]: Fixed: Removed hardcoded version numbers from the table and added a reference to `packages.config` instead, since versions are managed there and updated periodically.

File: .github/prompts/create-pr-summary.prompt.md [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+2. Run `git status --short` once to surface uncommitted files that may influence the summary.
+3. Run `git diff <target-branch>...HEAD` a single time to review the detailed changes. Only when confidence stays low dig deeper with focused calls such as `git diff <target-branch>...HEAD -- <path>`.
+4. From the diff, capture impacted WSL areas, key file changes, behavioral risks, migrations, and noteworthy edge cases. Use repo-specific component names when obvious (for example `service`, `wsl`, `init`, `gns`, `plan9`, `doc`, or `distributions`).
+5. Confirm validation: list tests or validation commands executed with results, or state why they were skipped in line with repo guidance.
+6. Load `.github/pull_request_template.md`, mirror its section order, and populate it with the gathered facts. Include only relevant checklist entries, such as tests, localization, dev docs, or docs repo follow-up, marking them `[x]/[ ]` and noting any intentional omissions as "N/A".
+7. Present the filled template inside a fenced code block opened with ````markdown``` with no extra commentary so it is ready to paste into a PR, clearly flagging any placeholders that still need user input.
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/prompts/create-pr-summary.prompt.md]: Fixed: Replaced the mismatched backtick fence delimiter with plain prose ('fenced Markdown code block') to avoid rendering issues.

File: .github/instructions/architecture.instructions.md [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+│                                                                    │
+│  C:\Windows\System32\wsl.exe ──CreateProcess()──► wsl.exe         │
+│                                                     │ COM         │
+│  wslg.exe ──────────────── COM ──────────────►  wslservice.exe    │
+│  wslapi.dll ──────────────── COM ──────────────►    │             │
+│    ▲ LoadLibrary()                                  │             │
+│  debian.exe, ubuntu.exe, ...            CreateProcessAsUser()     │
+│                                          ┌──────────┴──────────┐  │
+│  Windows filesystem                 wslrelay.exe    wslhost.exe   │
+│  (//wsl.localhost)                                                │
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/architecture.instructions.md]: Fixed: Changed `//wsl.localhost` to `\\wsl.localhost` UNC form for consistency with the rest of the document.

File: .github/copilot-instructions.md [hunk 10]
```diff
@@ -2,242 +2,217 @@
...
-4. **Distribution Validation**: Run Python validation scripts on any distribution changes
-
-### Manual Validation Requirements
-- **Windows builds**: Install MSI and test basic WSL functionality (`wsl --version`, `wsl -l`)
-- **Documentation changes**: Review generated HTML in `doc/site/`
-- **Distribution changes**: Test with actual WSL distribution installation
+| Resource | URL |
+|----------|-----|
+| User documentation | https://learn.microsoft.com/windows/wsl/ |
+| Developer documentation | https://wsl.dev/ |
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/copilot-instructions.md:12]: We just use single |, not double? I think nothing need to be fixed by this comment.

File: .github/instructions/agents.instructions.md [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+- Purpose: Define specialized agents with tailored expertise for WSL development
+
+## Example Frontmatter
+
+```yaml
+---
+description: 'Brief description of the agent purpose and capabilities'
+# name: 'Agent Display Name'      # Optional
+# tools: ['read', 'edit', 'search']  # Optional
+# model: 'Claude Sonnet 4.5'      # Recommended
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/agents.instructions.md]: Make this as "optional" rather than "Recommended", since it is very annoying for people who use different models there.

File: .github/instructions/instructions.instructions.md [hunk 12]
```diff
@@ -0,0 +1,256 @@
...
+- Verification steps
+
+## Content Guidelines
+
+### Writing Style
+
+- Use clear, concise language
+- Write in imperative mood ("Use", "Implement", "Avoid")
+- Be specific and actionable
+- Avoid ambiguous terms like "should", "might", "possibly"
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/instructions.instructions.md]: Fixed: Rephrased from banning the word 'should' to discouraging hedging words in imperative rules, resolving self-contradiction.

File: .github/copilot-instructions.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/copilot-instructions.md:12]: Not applicable: Tables already use single | pipes. Verified the rendered output is correct.

File: .github/instructions/documentation.instructions.md [hunk 13]
```diff
@@ -0,0 +1,40 @@
...
+## Published Docs
+
+- **User docs**: https://learn.microsoft.com/windows/wsl/
+- **Developer docs**: https://wsl.dev/
+- **Source**: `doc/` directory, built with MkDocs
+
+## Building Documentation (Linux or Windows)
+
+```bash
+pip install mkdocs-mermaid2-plugin mkdocs --break-system-packages
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/documentation.instructions.md]: Fixed: Reordered to python -m pip install --break-system-packages before package list.

File: .github/instructions/agents.instructions.md (hunk 11 above)
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/agents.instructions.md]: Fixed: Changed model field from Recommended to Optional per reviewer direction, since model availability varies by org.

File: .github/copilot-instructions.md [hunk 14]
```diff
@@ -2,242 +2,217 @@
...
 
-### Debug Console (Linux)
-Add to `%USERPROFILE%\.wslconfig`:
-```ini
-[wsl2]
-debugConsole=true
-```
+# Debug console (add to %USERPROFILE%\.wslconfig)
+# [wsl2]
+# debugConsole=true
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/copilot-instructions.md]: Fixed: Moved .wslconfig debug console example out of PowerShell block into a proper ini fenced block for easy copy-paste.

File: .github/instructions/agents.instructions.md [hunk 15]
```diff
@@ -0,0 +1,90 @@
...
+
+# Custom Agent File Guidelines — WSL
+
+Instructions for creating custom agent files that provide specialized expertise for WSL development tasks in GitHub Copilot.
+
+## File Basics
+
+- File format: Markdown with YAML frontmatter
+- Naming: lowercase with hyphens (e.g., `wsl-build-helper.agent.md`)
+- Location: `.github/agents/` directory
```
[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/agents.instructions.md]: Fixed: Added note to create .github/agents/ directory when adding the first agent file.

[the remark this answers is not in the document]
Comment (yeelam-gordon) [reply .github/instructions/agents.instructions.md]: That's fine, copilot does have this basic knowledge, don't need to add too many here.

**Comment by benhillis** [comment]
Can we be more iterative with this? This is just so gigantic I don’t want to read all this…
